lib/srv/db/common/auth.go

```diff
@@ -101,9 +100,10 @@ type Auth interface {
 	GetAzureIdentityResourceID(ctx context.Context, identityName string) (string, error)
 	// GetAWSIAMCreds returns the AWS IAM credentials, including access key,
 	// secret access key and session token.
-	GetAWSIAMCreds(ctx context.Context, sessionCtx *Session) (string, string, string, error)
-	// Closer releases all resources used by authenticator.
-	io.Closer
+	GetAWSIAMCreds(ctx context.Context, database types.Database, databaseUser string) (string, string, string, error)
+	// WithLogger returns a new instance of Auth with updated logger.
+	// The callback function receives the current logger and returns a new one.
+	WithLogger(getUpdatedLogger func(logrus.FieldLogger) logrus.FieldLogger) Auth
```

Comment:
nit: Instead of updating the existing logger, could we pass in the new logger? Or maybe `WithSessionLogger(sessionCtx)`?

Reply:
I think `WithSessionLogger` is too specific; I want to use that same function in the future in other contexts.

The current function is free to ignore the parameter and return a completely new logger, so it is more generic. In general, we have no way to inspect the logger `Auth` is using. It may have some custom fields set and we shouldn't just discard those, but we may add our own information on top.